File: delightreading-server/src/main/java/com/delightreading/common/HibernateStringListUserType.java
package com.delightreading.common;

import com.fasterxml.jackson.core.type.TypeReference;

import java.util.List;

public class HibernateStringListUserType extends JsonBasedType {
    public HibernateStringListUserType() {
        super(new TypeReference<List<String>>() {
        });
    }

    @Override
    public Class<List<String>> returnedClass() {
        return (Class<List<String>>) (Object) List.class;
    }
}

File: delightreading-backend/src/main/kotlin/com/delightreading/common/HibernateStringListUserType.kt
package com.delightreading.common

import com.fasterxml.jackson.core.type.TypeReference

class HibernateStringListUserType : JsonBasedType(object : TypeReference<List<String>>() {
}) {

    override fun returnedClass(): Class<List<String>> {
        return List::class.java as Class<List<String>>
    }
}